### PR TITLE
console: simplify console logging

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

@duguhaotian, this is my proposal.

### CODE REVIEW

File: src/lxc/console.c [hunk 1]
```diff
@@ -583,7 +746,7 @@ void lxc_console_delete(struct lxc_console *console)
 
 	ret = lxc_console_write_ringbuffer(console);
```
Comment (duguhaotian) [inline src/lxc/console.c]: Do we need this?  lxc_console_cb_con write console log into ringbuffer and log file, i think log file have the same logs with ringbuffer.
```
lxc_console_cb_con() {
     if (console->buffer_size > 0)
 			w_rbuf = lxc_ringbuf_write(&console->ringbuf, buf, r);
     if (console->log_fd > 0)
			w_log = lxc_console_write_log_file(console, buf, r);
}
```

Comment (brauner) [reply src/lxc/console.c]: > Do we need this? lxc_console_cb_con write console log into ringbuffer and log file, i think log file have the same logs with ringbuffer.

If `console->log_fd > 0` then the user wants us to write to the logfile. If they only care about the ringbuffer they can just choose to not use a console logfile.

Comment (brauner) [reply src/lxc/console.c]: Oh, sorry I misread your comment. The idea is that if the container e.g. aborts you get a dump of the last ringbuffer contents.

**Comment by stgraber** [comment]
@hallyn can you take a look at this one?

File: src/lxc/console.c [hunk 2]
```diff
@@ -204,6 +204,152 @@ void lxc_console_signal_fini(struct lxc_tty_state *ts)
...
+	return lxc_unpriv(ftruncate(console->log_fd, 0));
+}
+
+static int lxc_console_rotate_log_file(struct lxc_console *console)
+{
+	int ret;
+	size_t len;
+	char *tmp;
+
+	/* rotate the console log file */
```
Comment (hallyn) [inline src/lxc/console.c]: This comment seems superfluous

File: src/lxc/console.c [hunk 3]
```diff
@@ -204,6 +204,152 @@ void lxc_console_signal_fini(struct lxc_tty_state *ts)
...
+		 * he's requesting us to do weird stuff.
+		 */
+		if (console->log_rotate > 0 || console->log_size > 0)
+			return -EINVAL;
+
+		/* I mean, sure log wherever you want to. */
+		return lxc_write_nointr(console->log_fd, buf, bytes_read);
+	}
+
+	/* check how much space we have left */
```
Comment (hallyn) [inline src/lxc/console.c]: The useful variable name makes the comment unnecessary.

File: src/lxc/console.c [hunk 4]
```diff
@@ -701,10 +701,53 @@ static int lxc_console_peer_default(struct lxc_console *console)
...
+
+	/* There's not log file where we can dump the ringbuffer to. */
+	if (console->log_fd < 0)
+		return 0;
+
+	/* The log file is simply appended to. */
+	if (console->log_size == 0)
+		return 0;
+
+	/* The log file is rotated. */
```
Comment (hallyn) [inline src/lxc/console.c]: Sorry, why do you not dump the final contents on exit if the user does not want log files rotated?

File: src/lxc/console.c (hunk 2 above)
Comment (brauner) [reply src/lxc/console.c]: removed

File: src/lxc/console.c (hunk 3 above)
Comment (brauner) [reply src/lxc/console.c]: removed

File: src/lxc/console.c (hunk 4 above)
Comment (brauner) [reply src/lxc/console.c]: You're right, we should dump it.

**Pull request merged**